src/Api/Api.php

```diff
@@ -100,12 +117,13 @@ public function getFiscalInfo(): array
 
         $headers = $this->getEssentialHeaders();
 
-        $headers[TransferConstants::AUTHORIZATION_HEADER] = $this->token->getToken();
+        // $headers[TransferConstants::AUTHORIZATION_HEADER] = $this->token->getToken();
+        $headers['Authorization'] = 'Bearer ' . $this->token->getToken();
 
         return $this->httpClient->sendPacket('req/api/self-tsp/sync/GET_FISCAL_INFORMATION', $packet, $headers);
     }
 
-    public function setToken(Token $token): self
+    public function setToken(null|Token $token): self
```

Comment:
I don't think we need the property to be null by default or allow it to be set to null.
1. We are calling an endpoint that does not need authentication so the `getToken()` should never be called. In this case we don't need a default value for the token.
2. We are calling an endpoint that requires authentication in which case returning null from `getToken()` doesn't make anything better as we will run into errors later when calling the API instead of immediately.

Reply:
I agree, but `setToken` must not accept null anyway.

Reply:
Based on the code you wrote, "setToken" does not need to be called! I changed it to accept null values because the parent class ("Moadian") calls it forcefully. "setToken" and "getToken" need to be called if you have any caching system in place to improve performance.

You can test "getFiscalInfo" without calling "setToken" and "getToken" to see if it is working or not.